support-frontend/assets/pages/[countryGroupId]/oneTimeCheckout.tsx

```diff
@@ -176,7 +176,17 @@ function OneTimeCheckoutComponent({
 	const [otherAmount, setOtherAmount] = useState<string>('');
 	const [otherAmountError, setOtherAmountError] = useState<string>();
 
-	const finalAmount = amount === 'other' ? parseFloat(otherAmount) : amount;
+	/**
+	 * TODO: otherAmount falls back to zero if no other amount
+	 * entered by user. This prevents the function returning NaN.
+	 * Ideally it would return undefined and we'd handle that, but the
+	 * impact of doing so was deemed too great, considering most use cases
+	 * are in the soon to be deprecated 2-step checkout.
+	 */
```

Comment:
Not sure this comment feels relevant / accurate in this context as it was written in the context of code that will be deleted in the (hopefully near) future, whereas this is code that will persist.

Reply:
Actually have decided to pull changes to `oneTimeCheckout.tsx` completely. Hopefully this PR can go in. I will do the changes separately within the [ApplePay/GooglePay PR](https://github.com/guardian/support-frontend/pull/6351)